python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -1365,9 +1430,1084 @@ def get() -> "AssetExecutionContext":
+                    context.log.info(context.asset_partitions_time_window_for_input("self_dependent_asset"))
+
+                # materializing the 2023-08-21 partition of this asset will log:
+                #   TimeWindow("2023-08-20", "2023-08-21")
+
+                # running a backfill of the 2023-08-21 through 2023-08-25 partitions of this asset will log:
+                #   TimeWindow("2023-08-20", "2023-08-25")
+
+        """
+        return self.op_execution_context.asset_partitions_time_window_for_input(input_name)
+
+    @public
+    @experimental
+    def get_asset_provenance(self, asset_key: AssetKey) -> Optional[DataProvenance]:
+        """Return the provenance information for the most recent materialization of an asset.
+
+        Args:
+            asset_key (AssetKey): Key of the asset for which to retrieve provenance.
+
+        Returns:
+            Optional[DataProvenance]: Provenance information for the most recent
+                materialization of the asset. Returns `None` if the asset was never materialized or
+                the materialization record is too old to contain provenance information.
+        """
+        return self.op_execution_context.get_asset_provenance(asset_key=asset_key)
+
+    def set_data_version(self, asset_key: AssetKey, data_version: DataVersion) -> None:
+        """Set the data version for an asset being materialized by the currently executing step.
+        This is useful for external execution situations where it is not possible to return
+        an `Output`.
+
+        Args:
+            asset_key (AssetKey): Key of the asset for which to set the data version.
+            data_version (DataVersion): The data version to set.
+        """
+        return self.op_execution_context.set_data_version(
+            asset_key=asset_key, data_version=data_version
+        )
+
+    @property
+    def asset_check_spec(self) -> AssetCheckSpec:
+        return self.op_execution_context.asset_check_spec
+
+    @property
+    def is_subset(self):
+        return self.op_execution_context.is_subset
+
+    # In this mode no conversion is done on returned values and missing but expected outputs are not
+    # allowed.
+    @property
+    def requires_typed_event_stream(self) -> bool:
+        return self.op_execution_context.requires_typed_event_stream
+
+    @property
+    def typed_event_stream_error_message(self) -> Optional[str]:
+        return self.op_execution_context.typed_event_stream_error_message
+
+    def set_requires_typed_event_stream(self, *, error_message: Optional[str] = None) -> None:
+        self.op_execution_context.set_requires_typed_event_stream(error_message=error_message)
+
     @cached_method
     def get_op_execution_context(self) -> "OpExecutionContext":
-        return OpExecutionContext(self._step_execution_context)
+        # TODO - potentially delete in favor of self.op_execution_context if not a breaking change
```

Comment:
@alangenfeld  - what do you think about replacing `get_op_execution_context` with the  `op_execution_context` property?

Reply:
should be fine, this was recently introduced and i think fine to remove